### ISSUE TITLE
Parcel 2: .env files doesn't accept "_" in variable names, .env files not watched

### ISSUE THREAD

**Issue by Banou26** [parcel-bundler/parcel#3942]
# 🐛 bug report
Trying to use a `_` in a .env variable name result in the entire file env file not being used.
Side note, `.env` files aren't watched for changes.

## 🎛 Configuration (.babelrc, package.json, cli command)
`index.js`
```js
console.log(process.env.FOO_) // undefined
```
`.env`
```js
FOO_=BAR
```

**Comment by Banou26** [comment]
If the package is from a lerna monorepo, the .env file used is the one at the root of the lerna monorepo and not the sub package.
I don't know if that's the desired behavior but it seems weird to me since it remove configuration isolation from each packages in the monorepo.

**Comment by Banou26** [comment]
For my previous comment, should i open another issue ?
Actually have both of the behaviors i've described would be best, use both the .env at the top of the monorepo, and use the .env in each sub packages as it'd allow for both shared config, and configuration isolation.
Any opinions on that ?

**Comment by mischnic** [comment]
Yes, it would make sense to have env files in the packages overwrite env files further up.

**Comment by devongovett** [comment]
Partially fixed by #3942. You won't need to clear the cache anymore, but `.env` files still aren't in the graph so won't be watched in dev mode. Restarting Parcel should at least work though. We'll eventually get the env file in the graph too.
